src/libs/elektra/keyset.c

```diff
@@ -1546,6 +1545,26 @@ int f (KeySet *ks)
  * ksRewind(). When you set an invalid cursor ksCurrent()
  * is 0 and ksNext() == ksHead().
  *
+ * @section cursor_directly Using Cursor directly
+ *
+ * You can also use the cursor directly
+ * by initializing it to some index in the Keyset
+ * and then incrementing or decrementing it, to
+ * iterate over the keyset.
+ *
+ * @code
+void iterate(KeySet *ks)
```

Comment:
Can you put this into a separate example file so that it gets compiled/code formatted and so on?

Furthermore, isn't in most cases the `for` loop better? (Except if you need the end position of the cursor afterwards for some reason.) You can make an example for both for and while.

Btw. you should now also have the rights to add "ready to merge". Please always rerun the build job if it failed. We are working on it to make it more stable, please report the most common problems so that we can fix them first.

Reply:
> Can you put this into a separate example file so that it gets compiled/code formatted and so on?

Where do these examples go? And do you mean removing it completely from the doc?

Reply:
Look for `@snippet` in this file, you will find plenty of examples how to do it. examples will be a good directory, maybe call the file examples/ksIterate.c